src/main/java/org/jabref/gui/fieldeditors/LinkedFileViewModel.java

```diff
@@ -322,6 +331,168 @@ public void moveToDefaultDirectory() {
+                    List<String> subDirs = new ArrayList<>();
+                    Path parent = currentFilePath.get().getParent();
+
+                    // Find last directory containing the target filename
+                    while (parent != null && parent.getFileName() != null) {
+                        String dirName = parent.getFileName().toString();
+                        if (!isRootLikeDirectory(dirName)) {
+                            subDirs.addFirst(dirName);
+                        }
+                        parent = parent.getParent();
+                    }
+
+                    Path targetPath = target.getParent();
+
+                    for (String subDir: subDirs) {
+                        targetPath = targetPath.resolve(subDir);
+                    }
+                    targetPath = targetPath.resolve(fileName);
+
+                    // Create necessaries directories
+                    Files.createDirectories(targetPath.getParent());
+
+                    Files.move(currentFilePath.get(), targetPath);
+                    System.out.println("File moved successfully to: " + targetPath);
+                    linkedFile.setLink(targetPath.toString());
+
+                    // Clean up empty directories after moving the file
+                    cleanupEmptyDirectories(originalPath);
+
+                    // Update the menu item text after moving the file
+                    if (moveFileItem != null) {
+                        Platform.runLater(() -> updateMoveFileItemText(moveFileItem, target));
+                        Platform.runLater(() -> updateMoveAndRenameFileItemText(moveAndRenameFileItem, target));
+                    }
+                } catch (
+                        IOException e) {
+                    if (toRename) {
+                        dialogService.showErrorDialogAndWait(
+                                Localization.lang("Move file to directory and Rename"),
+                                Localization.lang("Could not move file '%0'. and Rename", currentFilePath.get().toString()),
+                                e);
+                    } else {
+                        dialogService.showErrorDialogAndWait(
+                                Localization.lang("Move file"),
+                                Localization.lang("Could not move file '%0'.", currentFilePath.get().toString()),
+                                e);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks if a directory name appears to be a root-level directory that should be excluded
+     * from the subdirectory structure preservation.
+     */
+    private boolean isRootLikeDirectory(String dirName) {
+        // Get the username
+        String user = preferences.getFilePreferences().getUserAndHost().split("-")[0];
+
+        // Create a set of root-like directories and include the user dynamically
+        Set<String> rootLikeDirs = new HashSet<>(Set.of(
```

Comment:
We are recreating the same set each time we check if the directory is root. This doesn't sound efficient.

Reply:
Yes it is not correct. I am moving it as static and final. It will be loaded once now.